src/decoder.ts

```diff
@@ -383,6 +382,72 @@ export class Decoder<A> {
     });
   }
 
+  /**
+   * Decoder for fixed-length arrays, aka Tuples.
+   *
+   * Supports up to 8-tuples.
+   *
+   * Example:
+   * ```
+   * tuple([number(), number(), string()]).run([5, 10, 'px'])
+   * // => {ok: true, result: [5, 10, 'px']}
+   * ```
+   */
+  static tuple<A>(decoder: [Decoder<A>]): Decoder<[A]>;
+  static tuple<A, B>(decoder: [Decoder<A>, Decoder<B>]): Decoder<[A, B]>;
+  static tuple<A, B, C>(decoder: [Decoder<A>, Decoder<B>, Decoder<C>]): Decoder<[A, B, C]>;
+  static tuple<A, B, C, D>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>]
+  ): Decoder<[A, B, C, D]>;
+  static tuple<A, B, C, D, E>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>]
+  ): Decoder<[A, B, C, D, E]>;
+  static tuple<A, B, C, D, E, F>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>, Decoder<F>]
+  ): Decoder<[A, B, C, D, E, F]>;
+  static tuple<A, B, C, D, E, F, G>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>, Decoder<F>, Decoder<G>]
+  ): Decoder<[A, B, C, D, E, F, G]>;
+  static tuple<A, B, C, D, E, F, G, H>(
+    decoder: [
+      Decoder<A>,
+      Decoder<B>,
+      Decoder<C>,
+      Decoder<D>,
+      Decoder<E>,
+      Decoder<F>,
+      Decoder<G>,
+      Decoder<H>
+    ]
+  ): Decoder<[A, B, C, D, E, F, G, H]>;
+  static tuple<A>(decoders?: Decoder<A>[]) {
```

Comment:
no need to make `decoders` optional here, since we know from the overload definitions that we'll always have at least a one-tuple as input. That lets you get rid of the `&& decoders` in a few lines.

Reply:
Thanks! Added in 247b86b.